manifests/remote.pp

```diff
@@ -1,40 +1,114 @@
 # == Define: opendnssec::tsig
 #
 define opendnssec::remote (
-  Optional[Variant[Tea::Ipv4, Tea::Ipv4_cidr]] $address4  = undef,
-  Optional[Variant[Tea::Ipv6, Tea::Ipv6_cidr]] $address6  = undef,
-  Optional[String]                             $tsig      = undef,
-  Optional[String]                             $tsig_name = undef,
-  Tea::Port                                    $port      = 53,
+  Optional[Variant[Tea::Ipv4, Tea::Ipv4_cidr]] $address4             = undef,
+  Optional[Variant[Tea::Ipv6, Tea::Ipv6_cidr]] $address6             = undef,
+  Optional[String]                             $tsig                 = undef,
+  Optional[String]                             $tsig_name            = undef,
+  Optional[String]                             $tsig_name_transfer_request = undef,
+  Optional[String]                             $tsig_name_transfer_provide = undef,
+  Optional[String]                             $tsig_name_notify_in  = undef,
+  Optional[String]                             $tsig_name_notify_out = undef,
+  Tea::Port                                    $port                 = 53,
 ) {
   include ::opendnssec
   $user               = $::opendnssec::user
-  $group              = $::opendnssec::user
+  $group              = $::opendnssec::group
   $manage_ods_ksmutil = $::opendnssec::manage_ods_ksmutil
   $enabled            = $::opendnssec::enabled
   $base_dir           = $::opendnssec::remotes_dir
 
   if ! $address4 and ! $address6 {
     fail("${name} must specify either address4 or address6")
   }
+
   if $tsig {
     if ! defined(Opendnssec::Tsig[$tsig]) {
       fail("${name}: Opendnssec::Tsig['${tsig}'] does not exist")
     }
-    if ! $tsig_name {
-      fail("${name}: you must define tsig_name when you deinfe tsig")
+    if $tsig_name {
+      $_tsig_name_transfer_request = $tsig_name
+      $_tsig_name_transfer_provide = $tsig_name
+
+    } elsif ! $tsig_name_transfer_request or ! $tsig_name_transfer_provide {
```

Comment:
thanks for the feedback. the reason i've put this in, is to be backward compatible with the current wat of working with '$tsig_name'. if this paramenter is set, this is used, otherwise '$tsig_name_transfer_request' and '$tsig_name_transfer_provide' are used.